### PR TITLE
Adds maliput_osm backend to the applications.

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

Related to https://github.com/maliput/maliput_osm/issues/21

## Summary
Adds `maliput_osm` backend to the applications.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by francocipollone** [comment]
It will fail in the ros build farm CI as `maliput_osm` isn't part of the ros build farm yet.

**Comment by francocipollone** [comment]
Also relies on https://github.com/maliput/maliput_osm/pull/20

**Comment by francocipollone** [comment]
@ros-pull-request-builder retest

**Comment by francocipollone** [comment]
@ros-pull-request-builder retest this please

**Comment by francocipollone** [comment]
@ros-pull-request-builder retest this please

File: test/tools_test.cc [hunk 1]
```diff
@@ -113,6 +113,34 @@ GTEST_TEST(CreateRoadNetwork, DragwayRoadNetwork) {
   EXPECT_NE(nullptr, dynamic_cast<const dragway::RoadGeometry*>(dut->road_geometry()));
 }
 
+class CreateMaliputOsmRoadNetworkTest : public ::testing::Test {
+ public:
+  static constexpr int kOverwrite{1};
+  static constexpr char MULTILANE_RESOURCE_ROOT[] = "MULTILANE_RESOURCE_ROOT";
```
Comment (agalbachicar) [inline test/tools_test.cc:119]: I think you want OSM_RESOURCE_ROOT, right?

I wonder why "MULTILANE_RESOURCE_ROOT" worked here.

**Review by agalbachicar — CHANGES_REQUESTED** [review]
PTAL.

File: test/tools_test.cc (hunk 1 above)
Comment (francocipollone) [reply test/tools_test.cc]: Good catch. Actually it wasn't being used. Addressed at [b18386e](https://github.com/maliput/maliput_integration/pull/124/commits/b18386e56823b9aac2129c87249352bd464ad279)

**Review by agalbachicar — APPROVED** [review]
LGTM

**Pull request merged**